tests/forms_tests/widget_tests/test_radioselect.py

```diff
@@ -70,27 +80,337 @@ def test_nested_choices(self):
+            None,
+            html=html,
+        )
+
+    def test_render_label_value(self):
+        """
+        If the value corresponds to a label (but not to an option value), none
+        of the options are selected.
+        """
+        html = """
+        <div>
+          <div>
+            <label><input type="radio" name="beatle" value="J">John</label>
+          </div>
+          <div>
+            <label><input type="radio" name="beatle" value="P">Paul</label>
+          </div>
+          <div>
+            <label><input type="radio" name="beatle" value="G">George</label>
+          </div>
+          <div>
+            <label><input type="radio" name="beatle" value="R">Ringo</label>
+          </div>
+        </div>
+        """
+        self.check_html(
+            self.widget(choices=self.beatles),
+            "beatle",
+            "Ringo",
+            html=html,
+        )
+
+    def test_render_selected(self):
+        """
+        Only one option can be selected.
+        """
+        choices = [("0", "0"), ("1", "1"), ("2", "2"), ("3", "3"), ("0", "extra")]
+        html = """
+        <div>
+          <div>
+            <label><input checked type="radio" name="choices" value="0">0</label>
+          </div>
+          <div>
+            <label><input type="radio" name="choices" value="1">1</label>
+          </div>
+          <div>
+            <label><input type="radio" name="choices" value="2">2</label>
+          </div>
+          <div>
+            <label><input type="radio" name="choices" value="3">3</label>
+          </div>
+          <div>
+            <label><input type="radio" name="choices" value="0">extra</label>
+          </div>
+        </div>
+        """
+        self.check_html(
+            self.widget(choices=choices),
+            "choices",
+            "0",
+            html=html,
+        )
+
     def test_constructor_attrs(self):
         """
         Attributes provided at instantiation are passed to the constituent
         inputs.
         """
-        widget = RadioSelect(attrs={"id": "foo"}, choices=self.beatles)
+        widget = self.widget(attrs={"id": "foo"}, choices=self.beatles)
         html = """
         <div id="foo">
+          <div>
+            <label for="foo_0">
+            <input checked type="radio" id="foo_0" value="J" name="beatle">John</label>
+          </div>
+          <div><label for="foo_1">
+            <input type="radio" id="foo_1" value="P" name="beatle">Paul</label>
+          </div>
+          <div><label for="foo_2">
+            <input type="radio" id="foo_2" value="G" name="beatle">George</label>
+          </div>
+          <div><label for="foo_3">
+            <input type="radio" id="foo_3" value="R" name="beatle">Ringo</label>
+          </div>
+        </div>
+        """
+        self.check_html(widget, "beatle", "J", html=html)
+
+    def test_compare_to_str(self):
+        """
+        The value is compared to its str()
+        """
```

Comment:
I don't see much value in this docstrings, it is the same at the test method name.

Reply:
This matches what's in `test_select`. To me, having the test structure, docstrings and their content (to some extent) being the same, makes more easy the diffing between `test_radioselect` and `test_select`. Even tough this branch is unifying some common tests, in the future, when a new test is added to one file, it'll likely be needed in the other side as well.

Reply:
I still don't see any much in such docstrings.